File: app/src/main/java/com/secondhand/vip/model/Product.kt
package com.secondhand.vip.model

import java.io.Serializable

data class Product(
    val _id: String,            // ★ 唯一 ID，對齊 MongoDB
    val name: String,
    val price: Int,
    val description: String,
    val imageUrls: List<String> = emptyList()
) : Serializable

File: app/src/main/java/com/secondhand/vip/model/Product.js
const mongoose = require("mongoose");

const ProductSchema = new mongoose.Schema(
  {
    name: { type: String, required: true },
    price: { type: Number, required: true },
    description: { type: String, default: "" },

    // ⭐ 舊資料

    imageUrl: { type: String, default: "" },

    // ⭐ 新：多圖
    imageUrls: { type: [String], default: [] }
  },
  { timestamps: true }
);

module.exports = mongoose.model("products", ProductSchema);
